tests/models/test_default_evaluator.py

```diff
@@ -499,3 +503,155 @@ def test_gen_multiclass_roc_curve():
 
     expected_auc = [0.75, 0.75, 0.3333]
     assert np.allclose(results.auc, expected_auc, rtol=1e-3)
+
+
+def test_evaluate_custom_metric_handle_none_result():
+    eval_df = pd.DataFrame({"prediction": [1.2, 1.9, 3.2], "target": [1, 2, 3]})
+    metrics = _get_regressor_metrics(eval_df["target"], eval_df["prediction"])
+
+    def dummy_fn(*_):
+        pass
+
+    assert _evaluate_custom_metric(dummy_fn, eval_df, metrics) == (None, None)
+
+
+def test_evaluate_custom_metric_incorrect_return_formats():
+    eval_df = pd.DataFrame({"prediction": [1.2, 1.9, 3.2], "target": [1, 2, 3]})
+    metrics = _get_regressor_metrics(eval_df["target"], eval_df["prediction"])
+
+    def incorrect_return_type_1(*_):
+        return 3
+
+    def incorrect_return_type_2(*_):
+        return "stuff", 3
+
+    def non_str_metric_name(*_):
+        return {123: 123, "a": 32.1, "b": 3}
+
+    def non_numerical_metric_value(*_):
+        return {"stuff": 12, "non_numerical_metric": "123"}
+
+    def non_str_artifact_name(*_):
+        return {"a": 32.1, "b": 3}, {1: [1, 2, 3]}
+
+    for test_fn in (
+        incorrect_return_type_1,
+        incorrect_return_type_2,
+        non_str_metric_name,
+        non_numerical_metric_value,
+        non_str_artifact_name,
+    ):
+        with pytest.raises(
+            MlflowException,
+            match=(
+                re.escape(
+                    f"Custom metric '{test_fn.__name__}' did not return in an expected format."
+                    f"The two acceptable return types are:"
+                    f"1. Dict[AnyStr, Union[int, float, np.number]: a dictionary of metrics"
+                    f"2. Tuple[Dict[AnyStr, Union[int, float, np.number]], Dict[AnyStr, Any]]: a"
+                    f"   dictionary of metrics and a dictionary of artifacts."
+                    f"For more, check out the docstring for mlflow.evaluate"
```

Comment:
We should only assert on the exact substring that we know will probably persist, like `did not return in an expected format.` Maybe the fn name also.